src/hermit.rs

```diff
@@ -115,4 +133,33 @@ mod tests {
         assert_eq!(*first_op,
                    Op::GitInit(PathBuf::from("/home/geoff/.hermit-config/shells/new-one")));
     }
+
+    #[test]
+    fn can_inhabit_existing_shell() {
+        let config = MockConfig {
+            root_path: PathBuf::from(".config"),
+            allowed_shell_names: vec!["default".to_owned(), 
+                "new-one".to_owned()],
+            current_shell: "default".to_owned(),
+        };
+        let mut hermit = hermit(&config);
+        let mut file_ops = FileOperations::rooted_at("/home/houngj"); 
+        hermit.inhabit_shell(&mut file_ops, "new-one");
+        let shell = hermit.current_shell().unwrap();
+        assert_eq!(shell.name, "new-one"); 
+    }
+    
+    #[test]
+    fn cant_inhabit_nonexisting_shell() {
+        let config = MockConfig {
+            root_path: PathBuf::from(".config"),
+            allowed_shell_names: vec!["default".to_owned()],
+            current_shell: "default".to_owned(),
+        };
+        let mut hermit = hermit(&config);
+        let mut file_ops = FileOperations::rooted_at("/home/houngj");
+        let res = hermit.inhabit_shell(&mut file_ops, "not-a-shell");
+        assert!(res.is_err());
+        assert_eq!(res.err().unwrap(), Error::ShellDoesNotExist);
```

Comment:
I know you're copying a pattern that I wrote originally, but I think we can express these two asserts more cleanly like so:
```rust
assert_eq!(res, Err(Error::ShellDoesNotExist));
```

Please change the usage in `cant_set_the_current_shell_to_a_nonexistent_shell` as well to this new pattern.